data/transactions/logic/eval.go

```diff
@@ -281,23 +304,33 @@ func (ep EvalParams) log() logging.Logger {
 
 type scratchSpace = [256]stackValue
 
-type evalContext struct {
+// EvalContext is the execution context of AVM bytecode.  It contains
+// the full state of the running program, and tracks some of the
+// things that the program has been done, like log message and inner
+// transactions.
+type EvalContext struct {
 	EvalParams
 
 	stack     []stackValue
 	callstack []int
-	program   []byte // txn.Lsig.Logic ?
-	pc        int
-	nextpc    int
-	err       error
-	intc      []uint64
-	bytec     [][]byte
-	version   uint64
-	scratch   scratchSpace
-
-	cost     int // cost incurred so far
-	logCalls int // number of log calls so far
-	logSize  int // log size of the program so far
+
+	subtxn *transactions.SignedTxn // place to build for tx_submit
+
+	// The transactions Performed() and their effects
+	Effects []transactions.SignedTxnWithAD
```

Comment:
I want to keep this close to subtxn, as they go together conceptually.  I also don't want this at the top because the first thing someone thinks when they read what an EvalContext is shouldn't be "oh, it can create transactions", it should "oh, it's the stack, the pc, the program bytes, etc"

Note that `Logs` is also exported, but of low importance.  `cost/Cost` might reasonably be exported soon.  I don't think that should control their position.